rask-engine/src/collide.rs

```diff
@@ -38,6 +38,41 @@ impl Collide for Collidable {
     }
 }
 
+/// calculate the axis aligned bounding box for a set of collidables
+pub fn calculate_aabb<'a, I: Iterator<Item = &'a Collidable>>(vals: I) -> AABox {
+    let (mut minx, mut maxx, mut miny, mut maxy) = (0.0, 0.0, 0.0, 0.0);
+    let mut first_iter = true;
+    let mut f = |points: &[Vec2]| {
+        for point in points {
+            if first_iter {
+                minx = point.x();
+                maxx = point.x();
+                miny = point.y();
+                maxy = point.y();
+                first_iter = false;
+            } else {
+                minx = f32::min(minx, point.x());
+                maxx = f32::max(maxx, point.x());
+                miny = f32::min(miny, point.y());
+                maxy = f32::max(maxy, point.y());
+            }
+        }
+    };
+    for val in vals {
+        match val {
+            Collidable::Point(v) => f(&[*v]),
+            Collidable::AABox(AABox { pos, size }) => f(&[*pos, *pos + *size]),
+            Collidable::RBox(RBox { pos, v1, v2 }) => {
+                f(&[*pos, *pos + *v1, *pos + *v2, *pos + *v1 + *v2])
```

Comment:
```rs
            Collidable::AABox(&AABox { pos, size }) => f(&[pos, pos + size]),
            Collidable::RBox(&RBox { pos, v1, v2 }) => {
                f(&[pos, pos + v1, pos + v2, pos + v1 + v2])
```
would be slightly shorter (but I'm not absolutely sure if this is possible)

Reply:
Also, in the `AABox` case we already know, that `pos.x < pos.x + size.x` etc.
So we have some redundant comparisons in the `f(&[*pos, *pos + *size])` call.
But the implementation would be a little bit ugly, so I don't know if it's worth...